src/main/java/com/mission2/racingcar/CarRacingService.java

```diff
@@ -0,0 +1,111 @@
+package com.mission2.racingcar;
+
+import java.util.*;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+public class CarRacingService {
+    public static final int MAX_CAR_COUNT = 5;
+    public static final int MAX_GAME_COUNT = 10;
+    public static final int INIT_SCORE = 1;
+    public static final int FORWARD = 4;
+    public static final String CAR_NAME_DELIMITER = ",";
+
+    public void game(String[] carNames, int gameCount) {
+        System.out.println("실행결과");
+        Race race = initRace(carNames, gameCount);
+        proceedGame(race);
+        printWinner(getWinners(race.getCars()));
+    }
+
+    /**
+     * 자동차 이름 문자열 길이를 확인한다.
+     */
+    public boolean checkCarNames(String[] carNames) {
+        long count = Arrays.stream(carNames).filter(car -> car.length() > MAX_CAR_COUNT).count();
+
+        return (count > 0) ? false : true;
+    }
+
+    /**
+     * 자동차 이름의 배열을 만든다.
+     */
+    public String[] getCarNames(String input) {
+        return Arrays.stream(input.split(CAR_NAME_DELIMITER)).map(String::trim).toArray(String[]::new);
```

Comment:
```suggestion
        return Arrays.stream(input.split(CAR_NAME_DELIMITER))
                .map(String::trim)
                .collect(Collectors.toList());
```

- stream 은 가독성을 위해 한줄에 하나씩 선언하는 것이 좋습니다 (다른 메서드도 같이 확인해 주세요 ㅎㅎ)
- Array보다 Collection API를 사용하는 것이 훨씬 이득인데요, 왜 그럴까요? :)

Reply:
저는 크기를 지정해야 메모리 성능에 좋다고 배웠는데요. 리스트가 사용하기는 편하지만 크기를 명확하게 알 수 있는 목록은 배열을 주로 사용했습니다. 그런데 여기서는 성능까지는 고려하지 않아도 될 것 같고, 배열보다 리스트의 장점이 더 많으니 잘 활용해봐야겠네요!

- [Array vs Collection API](https://stackoverflow.com/questions/6100148/collection-interface-vs-arrays/6105705)
  - Array는 초기화된 길이 변경이 불가능하고, null을 허용해야 하며, 스레드에 안전하지 않다. 
  - List는 초기화된 길이의 변경이 가능하고, 스레드로부터 안전하고, 다양한 Collection API를 제공한다.

- [Array vs ArrayList](https://velog.io/@humblechoi/%EC%9E%90%EB%A3%8C%EA%B5%AC%EC%A1%B0-Array-vs-ArrayList)
  - ArrayList 의 기본 생성 길이는 10 이다. element를 add 할 때 capacity가 배열의 길이와 같아지면 기존용량 + 기존용량/2 만큼 크기가 늘어난 배열에 기존의 배열을 copy해 준다.
  - ArrayList의 자동 크기 조정은 임시 배열을 사용하여 이전 배열의 요소를 새 배열로 복사하기 때문에 성능이 저하된다.
  - ArrayList는 Generics(제네릭스)를 통해 Type-Safety를 보장한다.